assets/js/common/OperationModals/ClusterMaintenanceChangeModal.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+import { concat, flatMap, flow, get, map, uniqBy } from 'lodash';
+import Label from '@common/Label';
+import Select from '@common/Select';
+import Switch from '@common/Switch';
+import OperationModal from './OperationModal';
+
+const NOT_SELECTED = 'Select a cluster resource';
+const CLUSTER_LABEL = 'Cluster (full maintenance)';
+const NODE_MAINTENANCE_STATE = 'Maintenance';
+const CLUSTER_SCOPE = 'cluster';
+const NODE_SCOPE = 'node';
+const RESOURCE_SCOPE = 'resource';
+
+const NOT_SELECTED_OPTION = {
+  value: {
+    id: NOT_SELECTED,
+    maintenance: false,
+  },
+  key: 'not_selected',
+};
+
+const renderOption = ({ value: { id } }) => id;
+
+function ClusterMaintenanceChangeModal({
+  clusterDetails,
+  isOpen = false,
+  onRequest,
+  onCancel,
+}) {
+  const [checked, setChecked] = useState(false);
+  const [resource, setResource] = useState(NOT_SELECTED_OPTION.value);
+  const [maintenanceState, setMaintenanceState] = useState(false);
+
+  const clusterMaintenance = get(clusterDetails, 'maintenance_mode', false);
+  const clusterNodes = get(clusterDetails, 'nodes', []);
+  const stoppedResource = get(clusterDetails, 'stopped_resources', []);
+
+  const clusterOption = {
+    value: {
+      id: CLUSTER_LABEL,
+      maintenance: clusterMaintenance,
+      scope: CLUSTER_SCOPE,
+    },
+    key: 'cluster',
+  };
+
+  const nodeOptions = map(clusterNodes, ({ name, status }) => ({
+    value: {
+      id: name,
+      maintenance: status === NODE_MAINTENANCE_STATE,
+      scope: NODE_SCOPE,
+    },
+    key: name,
+  }));
+
+  // flatMap all resources and parents from all nodes
+  // and concat stopped resources
+  const resourceOptions = flow([
+    (nodes) => flatMap(nodes, ({ resources }) => resources),
+    (resources) =>
+      flatMap(resources, (res) => (res.parent ? [res.parent, res] : [res])),
+    (resources) => concat(resources, stoppedResource),
+    (resources) => uniqBy(resources, 'id'),
+    (resources) =>
+      map(resources, ({ id, managed }) => ({
+        value: { id, maintenance: !managed, scope: RESOURCE_SCOPE },
+        key: id,
+      })),
+  ])(clusterNodes);
+
+  const allOptions = concat(
+    NOT_SELECTED_OPTION,
+    clusterOption,
+    nodeOptions,
+    resourceOptions
+  );
+
+  return (
+    <OperationModal
+      title="Cluster Maintenance"
```

Comment:
i guess this could also be a const variable label put on top?

Reply:
also goes for description and  operationText?

Reply:
Let's keep as it is. They are not used anywhere else and are just text that don't have any other meaning